### DISCUSSION TITLE
피자 상세 정보에 어떤 정보를 추가하는게 좋을까요?!

### DISCUSSION THREAD

**Discussion by TedLeem** [cau-swe/checkpoint-1#26] [Ideas]
사용자가 피자를 선택할 때, 피자 상세페이지에서 해당 피자의 관련 정보들을 보고 장바구니에 담을지 결정하는 "메뉴 선택" 기능을 Customer쪽에서 추가했습니다. 이때, 피자에 대한 상세정보들은 관리자쪽에서 추가해야 하는 문제인 것 같아서 피자 상세정보에 피자의 어떤 정보가 포함되면 좋을지 같이 정하면 좋을 것 같아 의견을 물어보겠습니다!
상세정보를 간단하게 다음과 같이 정했는데 혹시 더 추가되면 좋겠으면 하는 정보가 있으면 편하게 의견내주시면 감사하겠습니다. 
### 피자상세 정보
- 피자이름
- 피자가격
- 피자후기들

**Comment by Topslug** [comment]
저는 인터페이스가 어떻게 되느냐에 따라 달라질 것 같습니다.
Display하는 정보들이 어떻게 되느냐에 따라 달라질 것 같은데 만약 피자이름, 피자가격이 한번에 보여진다면, 피자 상세정보는 Review만이 될 것 같습니다.
그리고 처음 보고 생각이 들었던 것은 상세정보로 어떠한 재료가 들었는지(ex 맛있는 새우와 버터와의 조합이 환상적입니다!), 칼로리 같은 것이 있는데, 이러한 정보들도 같이 display될 수 있지 않을까 싶습니다.
만약 그렇다면 Review를 볼 수 있는 인터페이스만 고려하면 될 것 같습니다.

피자보기 Use Case가 완료되면 이 정도의 화면구성이 보여지는 것으로하면 어떨지 제안드립니다.

![피자메뉴화면](https://user-images.githubusercontent.com/53007295/114793297-03096e80-9dc5-11eb-9d29-05ab107d4df2.jpg)

**Reply by Topslug** [reply]
그리고 여기서 Review옆에 장바구니 담기와 같은 인터페이스도 추가되면 되지 않을까요?
그러면 피자선택이라는 Use Case는 따로 안 만들어도 되지 않을까 싶습니다. 왜냐하면 피자선택이라는 말은 곧 장바구니 담기 또는 Review보기 이니까요.
use case 메뉴보기
use case 장바구니 담기
use case Review보기
만 있어도 충분할 것으로 보입니다.

음.. 그리고 이건 개인적인 생각인데, diagram상에 extend를 쓰는 것은 안 하셔도 되지 않을까 싶습니다.
제가 Generalization에 관해 교수님께 질문드렸을 때 들었던 생각은 UC-1과 UC-2를 UC-3으로 Generalization을 하면 곧 UC-3만을 가지고 UC-1과 UC-2를 표현할 수 있게 되는 것이고, UC-1은 UC-3이다라고 할 수 있게 된다라는 것입니다.
그런데 메뉴보기 extend review보기를 봤을 때, review보기는 곧 메뉴보기이다라는 말은 성립하지 않습니다.
저는 교수님께서 Generalization을 언급하시면서 extend를 설명해주셨다라는 맥락을 볼 때, extend는 generalizaiton의 경우에만 쓰는 것이라고 생각을 해서, extend는 안 써도 될 것 같습니다.
그리고 메뉴보기라는 use case가 extend된 리뷰보기를 수행해야만 완료되는 use case가 아니므로 extend를 쓰는 것은 조금 이상한 것 같습니다.(use case ManageUsers에서 extend된 use case Add Users에 왜 스키마상에서는 include를 썼는지 참고하시면 될 것 같습니다.)

**Reply by Topslug** [reply]
아 그렇다면 장바구니 담기와 review보기는 메뉴선택으로 generalizaiton될 수 있으므로 이 부분은 어떻게 잘 표현되면 좋을 것 같습니다.

use case 메뉴보기
use case 장바구니 담기
use case Review보기
장바구니 담기 , Review보기 --> 메뉴선택
use case 메뉴선택

여기서 핵심은 세분화된 Use case를 작성 후 나중에 일반화를 하는 것이지 않나 싶습니다.

음.. 그런데 메뉴선택이라는 말이 주문을 위해 선택한다는 느낌이 좀 더 강한 것 같아서 굳이 이렇게 까지 할 필요는 없을 것 같기도 합니다.

아니면 
Use Case 바로 구매와 
Use Case 장바구니 담기를 Generalization해서
Use Case 메뉴 선택이라고 해도 되지 않을까요?
Review에 남겨주신 걸로 봤을 때, 바로 구매를 없앴다고 하셨는데, 제 생각에는 이런 식으로 표현하면, 괜찮을 것 같습니다.

그리고 장바구니 담기를 완료하고 구매하기 버튼을 누르거나
아니면 바로 구매 버튼을 누르면, 그 이후로는 주문 정보 입력 과정, 주문이 완성되기 까지는 공통되므로 이 부분에 대해서 generalization된 use case를 include하여 쉽게 표현할 수 있을 것 같습니다. Generalization된 Use Case는 Domain Model을 작성할 필요가 없으므로 한 번 고려해보셔도 좋을 것 같습니다.
바로 구매가 꽤 편리한 기능이라 없앤 것이 조금 아쉽긴 하네요.

**Reply by Topslug** [reply]
다시 생각해보니 바로 구매를 없애고 장바구니 구매만 선택한 것은 탁월한 선택이었던 것 같습니다

피자 상세 정보를 피자정보 + Review정보라고 한다면
Review보기를 클릭했을 때 새로운 페이지가 나오면서
피자정보와 그 밑에 Review를 보여줘도 될 것 같습니다.. 제가 너무 용어를 자세하게 나눌려고 했던 것 같습니다. 죄송합니다.
